packages/query-devtools/src/utils.tsx

```diff
@@ -111,3 +111,56 @@ export const convertRemToPixels = (rem: number) => {
 }
 
 export const convertPixelsToRem = (px: number) => px / convertRemToPixels(1)
+
+/**
+ * updates nested data by path
+ *
+ * @param {unknown} oldData Data to be updated
+ * @param {Array<string>} updatePath Path to the data to be updated
+ * @param {unknown} value New value
+ */
+export const updateNestedDataByPath = (
+  oldData: unknown,
+  updatePath: Array<string>,
+  value: unknown,
+): any => {
+  if (updatePath.length === 0) {
+    return value
+  }
+
+  // @ts-ignore
```

Comment:
if we have to keep these, please prefer `@ts-expect-error`, because it will type-error if there is nothing to ignore.

Reply:
I updated the comments to use `@ts-expect-error` for now. 

We should probably come back to the issue @ardeora raised earlier though:

> A query function could also return a Map, Set, Iterable, etc.

Currently the two util functions are only covering arrays and objects properly. How can we handle other instaces? Can't find a good solution right now. Any idea?